packages/js/components/src/split-dropdown/split-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { cloneElement, createElement } from '@wordpress/element';
+import { Icon, chevronDown, chevronUp } from '@wordpress/icons';
+import { Button, ButtonGroup, Dropdown } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+export type SplitDropdownProps = {
+	className?: string;
+	disabled?: boolean;
+	variant?: string;
+	children: JSX.Element[];
+};
+
+export const SplitDropdown: React.FC< SplitDropdownProps > = ( {
+	className = '',
+	children,
+	disabled,
```

Comment:
How about instead of passing `disabled` we just have a `toggleProps` that gets passed to the toggle button?